gateway-messages/src/tlv.rs

```diff
@@ -0,0 +1,228 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Extremely simple / minimal implementation of tag/length/value encoding,
+//! using 4-byte tags and lengths.
+
+use core::iter;
+use core::mem;
+use zerocopy::byteorder::LittleEndian;
+use zerocopy::byteorder::U32;
+use zerocopy::AsBytes;
+use zerocopy::FromBytes;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum EncodeError<E> {
+    /// The output buffer is too small to contain a tag/length header.
+    BufferTooSmall,
+    /// Custom error, defined and provided by the caller.
+    Custom(E),
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum DecodeError {
+    /// The buffer is too small to contain a tag/length header.
+    BufferTooSmall,
+    /// The `length` field requires more data than is remaining in the buffer.
+    LengthTooLong,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, AsBytes, FromBytes)]
+#[repr(C)]
```

Comment:
Just curios, why the `repr(C)` here?

Reply:
It's a requirement to derive the zerocopy traits: https://docs.rs/zerocopy/latest/zerocopy/trait.AsBytes.html#safety